File: problems/problems_1295/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.findNumbers(test_input)

    def findNumbers(self, nums: List[int]) -> int:
        def count_digits(n: int) -> int:
            count = 0
            while n > 0:
                n //= 10
                count += 1
            return count

        return sum(count_digits(num) % 2 == 0 for num in nums)

File: problems/problems_1295/solution.go
package problem1295

import (
	"encoding/json"
	"log"
	"strings"
)

func findNumbers(nums []int) (ans int) {
	countDigits := func(n int) (count int) {
		for n > 0 {
			n /= 10
			count++
		}
		return
	}
	for _, num := range nums {
		if countDigits(num)%2 == 0 {
			ans++
		}
	}
	return
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}

	return findNumbers(nums)
}
